### ISSUE TITLE
Importing an API still puts it under `resources/`

### ISSUE THREAD

**Issue by BigBlueHat** [stoplightio/studio#300]
**Describe the bug**
Issue #57 removed the requirement for API docs (et al) to be stored in a `resources/` folder. However, importing an API from within the current working directory will still move it under `resources/` requiring the developer to then move it back out of `resources/`.

**To Reproduce**

1. Create a project from an existing folder containing one or more `.yaml` files
  - The files will appear under `Files` but not under `API`
2. Use the `Import` feature to add a `.yaml` file from the projects working directory root into Studio as APIs
3. A new `resources/` folder will be created and the `.yaml` file moved there
4. Be sad that it couldn't stay where you wanted it
5. Move it back

**Expected behavior**
If an API spec document is already within the working directory for the project, it should stay where it is and not be moved/reorganized.

Thanks for considering this one!
🎩

**Comment by BigBlueHat** [comment]
Related to this, loading an existing folder full of OpenAPI specs in YAML format requires a developer to go through the import step for each YAML file...and move them back out of the newly created `resources/` folder.

It would be fabulous to have Stoplight either detect the `.yaml` files which are API specs, or allow the developer to "toggle" them to be treated as APIs...or something with fewer steps...at least.

**Comment by lottamus** [comment]
Hey @BigBlueHat,

We are currently working on adding support for configuring the default directory for new files. You'll be able to configure the directory based on the file's type (openapi, markdown, etc). For example, right now the hardcoded default for "openapi" files is the "reference" directory. Furthermore, you'll be able to configure which directories and files paths are included/excluded from certain sidebar trees (APIs, Docs, Files, etc).

Stay tuned!

**Comment by philsturgeon** [comment]
Project Config does not exist meaning it can be read from anywhere, but this issue was initially about it being put _somewhere_. For now, yeah, Import is going to put them there.

Where should they be put? In the root? That is probably not where anyone really wants them if they're mixing code and descriptions. Any ideas?

**Comment by philsturgeon** [comment]
I was totally wrong above, the config is respected for placing new APIs somewhere too, not just reading them. This is all done, all ready to go, can be used in Studio Web or Studio Desktop v1.11.0 @BigBlueHat so check out this PR and let me know if you have any trouble. https://github.com/stoplightio/studio/pull/348

**Comment by BigBlueHat** [comment]
Doesn't look like the new config file stuff handles API docs in the current directory.

I tried:
```json
{
  "formats": {
    "openapi": {
      "rootDir": "."
    }
  }
}
```
which hid everything after I applied the config...

I then changed `rootDir` to `/`...which turned out to be invalid, but when re-opening the project my old setup was there.

Now, I can see this error in the config parser:
> rootDir must be a valid directory name with at least one character other than a slash

So, I'm not sure if it's possible to set the current directory as the home of the APIs (and doesn't seem to be noted in the docs).

In the end, I have several repositories (from others) that only (currently) contain OAS YAML files in the root (other than a project README.md). I'd like to be able to more easily set those up and manage them in Studio...in the end.

**Comment by P0lip** [comment]
@BigBlueHat 
Hey! Could you please try the following config?
```json
{
  "formats": {
    "openapi": {
      "rootDir": ".",
      "include": ["**"],
    },
  },
}
```

**Comment by BigBlueHat** [comment]
@P0lip that did the trick! But isn't include optional? Might also want to tweak those error messages in parsing. And, without the `include` bit the project files are all hidden (per above).

**Comment by P0lip** [comment]
> that did the trick!

Awesome!

> But isn't include optional? Might also want to tweak those error messages in parsing. And, without the include bit the project files are all hidden (per above).

It's optional for regular directories, but not for the project root (`.`).
In general the idea behind Stoplight config was to instrument projects in our ecosystem as to which files are relevant, and which are less so.
Using the project root as `rootDir` kind of violates that purpose, as it literally stands for "all files are equally important" and that's why `include` is needed.
I know it's tad confusing, so perhaps will make it more aligned with regular directories to make sure `include` is no longer required in such a case.
I hope that makes sense.

**Comment by philsturgeon** [comment]
Hey @lordlycastle, pretty sure you are experiencing the same trouble that was discovered over here https://github.com/stoplightio/studio/issues/368 The issue has been resolved and it'll come out in a version post-v1.11, but for now try shoving some / on the front. If that doesn't work please make a new issue, as this is a different thing.

**Comment by philsturgeon** [comment]
Btw I locked this one bc I can forsee this becoming a general "stoplight config support thread". 😅
